src/main/java/uk/gov/pay/adminusers/service/InviteRouter.java

```diff
@@ -25,30 +26,42 @@ public Optional<InviteCompleter> routeComplete(String inviteCode) {
                     switch (inviteEntity.getType()) {
                         case SERVICE:
                         case NEW_USER_AND_NEW_SERVICE_SELF_SIGNUP:
-                            return Optional.of(inviteServiceFactory.completeSelfSignupInvite());
+                            return inviteServiceFactory.completeSelfSignupInvite();
                         case USER:
                         case EXISTING_USER_INVITED_TO_EXISTING_SERVICE:
-                            return Optional.of(inviteServiceFactory.completeExistingUserInvite());
+                            return inviteServiceFactory.completeExistingUserInvite();
                         case NEW_USER_INVITED_TO_EXISTING_SERVICE:
-                            return Optional.of(inviteServiceFactory.completeNewUserExistingServiceInvite());
+                            return inviteServiceFactory.completeNewUserExistingServiceInvite();
                         default:
                             throw new IllegalArgumentException(String.format("Unrecognised invite type: %s", inviteEntity.getType()));
                     }
                 });
     }
 
+    /**
+     * @return an optional pair consisting of: the InviteOtpDispatcher to use, and a boolean flag to indicate whether the OTP requires validation during invite completion.
```

Comment:
I think the boolean flag is actually returning whether or not the invite type is user or service, and then using that to determine what it should validate on the request payload.

The code for that is horrible and we can do some work later down the line to tidy that up. However for now, would it make things a tiny bit clearer to return a Pair<InviteOtpDispatcher, InviteType> - it should make the refactor a bit easier later on.